From f44064fdb96386943da4592fc8bf7ae5cf42637c Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:49 -0400
Subject: [virt] kvm/x86: mark hyper-v hypercall page as dirty

Message-id: <1406040016-3289-135-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86213
O-Subject: [RHEL7 PATCH v2 134/221] KVM: x86: mark hyper-v hypercall page as dirty
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Vadim Rozenfeld <vrozenfe@redhat.com>

Signed-off-by: Vadim Rozenfeld <vrozenfe@redhat.com>
Reviewed-by: Marcelo Tosatti <mtosatti@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit b94b64c9a79a322802f13f44e9690a7b6a22710e)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 9b28850..86f9c8d 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -1873,6 +1873,7 @@ static int set_msr_hyperv_pw(struct kvm_vcpu *vcpu, u32 msr, u64 data)
   if (__copy_to_user((void __user *)addr, instructions, 4))
    return 1;
   kvm->arch.hv_hypercall = data;
+  mark_page_dirty(kvm, gfn);
   break;
  }
  case HV_X64_MSR_REFERENCE_TSC: {
-- 
1.7.1